shop4All-backend/src/main/java/com/example/shop4All_backend/configurations/WebSecurityConfiguration.java

```diff
@@ -44,10 +44,11 @@ public SecurityFilterChain configure(HttpSecurity http) throws Exception {
                 // Disable CSRF protection
                 .csrf(AbstractHttpConfigurer::disable)
                 // Configure authorization rules for different endpoints
-                .authorizeHttpRequests(auth -> {
-                    auth.requestMatchers("/registerNewSeller", "/registerNewBuyer", "/authenticate", "/login", "/", "/product/getAll", "/product/getById/**", "category/getAll").permitAll();
-                    auth.anyRequest().authenticated();
-                })
+                    .authorizeHttpRequests(auth -> {
+                        auth.requestMatchers("/registerNewSeller", "/registerNewBuyer", "/authenticate", "/login", "/").permitAll();
+                        auth.requestMatchers("/product/getAll", "/product/getById/**", "category/getAll").permitAll();
+                        auth.anyRequest().authenticated();
+                    })
```

Comment:
Note: I think you can also add in the "requestMatchers" method the ["hasRole"](https://docs.spring.io/spring-security/reference/servlet/authorization/authorize-http-requests.html#authorize-requests) method like this:

`.requestMatchers("/admin/**").hasRole("ADMIN") `

This way it would be easier to navigate through your code and see which role can access the speficied endpoints.
If you want to apply this, check if you can delete @PreAuthorize annotation.